.github/workflows/publish.yaml

```diff
@@ -0,0 +1,50 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Get tags
+        id: get_tag
+        run: echo ::set-output name=TAG::${GITHUB_REF#refs/tags/}
+      - name: Get Owner name
+        id: get_owner_name
+        run: echo ::set-output name=OWNER_NAME::$(echo "$GITHUB_REPOSITORY" | awk -F / '{print tolower($1)}' | sed -e "s/:refs//")
+      - name: Get Repository name
+        id: get_repo_name
+        run: echo ::set-output name=REPOSITORY_NAME::$(echo "$GITHUB_REPOSITORY" | awk -F / '{print tolower($2)}' | sed -e "s/:refs//")
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
```

Comment:
I need some guidance on what this secret name _should_ be

Reply:
Is CR_PAT automated? If not look at our build for faas-netes

Reply:
The automatic value is `GITHUB_TOKEN` but there might be issues with using this to verify the publish flow in forks because it might only have read access, to be frank the docs are not very great about this use case

https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow

Reply:
Per these docs it worked for github packages https://docs.github.com/en/free-pro-team@latest/actions/guides/publishing-docker-images#publishing-images-to-github-packages and it is the docs that provide for github actions to publish docker images, i was hoping that they didn't deprecate that feature